pkg/controller/daemon/daemoncontroller.go

```diff
@@ -425,11 +425,28 @@ func (dsc *DaemonSetsController) addNode(obj interface{}) {
 	}
 }
 
+func compareConditions(oldConditions, curConditions []v1.NodeCondition) bool {
+	if len(oldConditions) != len(curConditions) {
+		return false
+	}
+	for i := range oldConditions {
+		old := oldConditions[i]
+		cur := curConditions[i]
+		if old.Type != cur.Type || old.Status != cur.Status {
+			return false
+		}
+	}
+	return true
+}
+
 func (dsc *DaemonSetsController) updateNode(old, cur interface{}) {
 	oldNode := old.(*v1.Node)
 	curNode := cur.(*v1.Node)
-	if reflect.DeepEqual(oldNode.Labels, curNode.Labels) && reflect.DeepEqual(oldNode.Spec.Taints, curNode.Spec.Taints) {
-		// If node labels and taints didn't change, we can ignore this update.
+	if reflect.DeepEqual(oldNode.Labels, curNode.Labels) &&
+		reflect.DeepEqual(oldNode.Spec.Taints, curNode.Spec.Taints) &&
+		reflect.DeepEqual(oldNode.Status.Allocatable, curNode.Status.Allocatable) &&
```

Comment:
I was thinking more about situations where the kubelet process would be restarted with a different `--max-pods` or `--system-reserved` flags. Also, some virtual machine providers support 'hot plugging' more memory/cpu to a live machine, requiring only a kubelet restart to have these values updated.
However, if it's guaranteed that the node's `Ready` condition will flip between True and False on every kubelet restart, then there's no need for this. But I wasn't able to verify that this is always the case.